### PR TITLE
Sync the two IDs in OrtMemoryInfo when calling ctor

### PR DESCRIPTION
**Pull request by wschin**
As title. `OrtMemoryInfo.id` and `OrtMemoryInfo.device.Id()` should be the same.

I think we should merge these two fields. Maybe you can create an item @SherlockNoMad for core?

### CODE REVIEW

**Review by snnn — APPROVED** [review]

File: onnxruntime/python/onnxruntime_pybind_state.cc [hunk 1]
```diff
@@ -1289,7 +1289,7 @@ void addObjectMethods(py::module& m, Environment& env) {
         int type_num = dtype->type_num;
         Py_DECREF(dtype);
 
-        OrtMemoryInfo info(GetDeviceName(device), OrtDeviceAllocator, device);
+        OrtMemoryInfo info(GetDeviceName(device), OrtDeviceAllocator, device, device.Id());
```
Comment (KeDengMS) [inline onnxruntime/python/onnxruntime_pybind_state.cc:1292]: >device.Id() [](start = 78, length = 11)

Seems like a bug that would be caught with multi-GPU CI. Some following up questions:
1. Should we change the OrtMemoryInfo ctor to not take default arguments of device_id == 0?
2. Do we have test coverage on this?

Comment (wschin) [reply onnxruntime/python/onnxruntime_pybind_state.cc:1292]: I think we should merge these two fields.

---
In reply to: [516222469](https://github.com/microsoft/onnxruntime/pull/5663#discussion_r516222469) [](ancestors = 516222469)

Comment (snnn) [reply onnxruntime/python/onnxruntime_pybind_state.cc:1292]: I have a multi-GPU daily pipeline. But I guess it didn't cover this code.

Comment (KeDengMS) [reply onnxruntime/python/onnxruntime_pybind_state.cc:1292]: @liqunfu Do we have CI tests in Python that uses multi-GPU?

Comment (jywu-msft) [reply onnxruntime/python/onnxruntime_pybind_state.cc:1292]: @hectorsvc set up a multi gpu inference test pipeline in the past.

Comment (snnn) [reply onnxruntime/python/onnxruntime_pybind_state.cc:1292]: Yes. https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=136907&view=results

**Pull request merged**